fbw-a32nx/src/systems/instruments/src/MCDU/legacy_pages/A320_Neo_CDU_AvailableArrivalsPage.ts

```diff
@@ -609,19 +607,69 @@ export class CDUAvailableArrivalsPage {
       selectedStarCellColor = planColor;
     }
 
-    const rows = [[''], [''], [''], [''], [''], ['']];
+    const rows = [[''], [''], [''], [''], [''], [''], [''], ['']];
+
+    const starFixes = [];
+    if (targetPlan.arrivalEnrouteTransition?.legs.length > 0) {
+      for (const leg of targetPlan.arrivalEnrouteTransition.legs) {
+        if (leg.waypoint) {
+          starFixes.push(leg.waypoint.databaseId);
+        }
+      }
+    }
+    if (targetPlan.arrival?.commonLegs.length > 0) {
+      for (const leg of targetPlan.arrival.commonLegs) {
+        if (leg.waypoint) {
+          starFixes.push(leg.waypoint.databaseId);
+        }
+      }
+    }
+    if (targetPlan.arrivalRunwayTransition?.legs.length > 0) {
+      for (const leg of targetPlan.arrivalRunwayTransition.legs) {
+        if (leg.waypoint) {
+          starFixes.push(leg.waypoint.databaseId);
+        }
+      }
+    }
+
+    const shouldFilterVias = starFixes.length > 0;
+    const vias: { ident: string; databaseId: string | null; isOtherVia: boolean }[] =
+      targetPlan.availableApproachVias.map((t) => ({
+        ident: t.ident,
+        databaseId: t.databaseId,
+        isOtherVia: shouldFilterVias && !starFixes.find((id) => t.legs[0].waypoint?.databaseId === id),
+      }));
```

Comment:
im gonna be that person and
```suggestion
    }));
```
👺

Reply:
The formatting is decided by prettier.